portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,40 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String messagesJson = convertToJson(messages);
     response.setContentType("text/html;");
-    response.getWriter().println("Hello Ramya!");
+    response.getWriter().println(messagesJson);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment = request.getParameter("text-input");
+    if (!comment.isEmpty()){ 
```

Comment:
according to the quick limited documentation I went through, if the parameter doesn't exist, it will return null. So in this case, this might evaluate into `null.isEmpty()`, which would be a NullPointerException.

You should check for null-ness before checking empty-ness.